app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java

```diff
@@ -251,4 +246,14 @@ public void startScan() {
     public void stopScan() {
         cameraSource.stop();
     }
+
+    public void showToggleDialog(long attendeeId) {
+        BottomSheetDialogFragment bottomSheetDialogFragment = AttendeeCheckInFragment.newInstance(attendeeId);
+        bottomSheetDialogFragment.show(getSupportFragmentManager(), bottomSheetDialogFragment.getTag());
+    }
+
+    @Override
+    public void onDialogDismissed() {
+        presenter.allowScan(true);
```

Comment:
There are two ways, one is like yours, just simpler, use a boolean as a filter and just change its value in the presenter using a method, it'll automatically stop and resume. But there is no need to apply conditionals and all. Just change boolean to false when dialog is shown and to true when it is hidden